src/main/java/io/boomerang/service/FlowExecutionServiceImpl.java

```diff
@@ -290,29 +271,26 @@ private void executeWorkflowAsync(String activityId, final Task start, final Tas
 
     final Task startTask = tasksToRun.stream()
         .filter(tsk -> TaskType.start.equals(tsk.getTaskType())).findAny().orElse(null);
-    executeNextStep(activityEntity, tasksToRun, startTask, start, end, graph);
+    executeNextStep(tasksToRun, startTask, start, end, graph);
   }
 
-  private void executeNextStep(ActivityEntity workflowActivity, List<Task> tasks, Task currentTask,
+  private void executeNextStep(List<Task> tasks, Task currentTask,
       final Task start, final Task end, final Graph<String, DefaultEdge> graph) {
 
-    try {
-      List<Task> nextNodes = this.getTasksDependants(tasks, currentTask);
-      for (Task next : nextNodes) {
+    if (currentTask == null || graph == null)
+      throw new IllegalArgumentException();
```

Comment:
Hi @AndreiOla where would the IllegalArgumentException be handled? do we not have to mark the method as throwing one of these or is it different for standard exceptions?

Reply:
@tlawrie This will be handled just like a NPE (RuntimeException), on this line (305) `return CompletableFuture.supplyAsync(createProcess(activityId, tasks));` But it will be more clear what might have caused it.